### ISSUE TITLE
Auto-import on build.sbt save

### ISSUE THREAD

**Issue by kubukoz** [scalameta/metals#432]
I'd love to have auto-import on every save in my `build.sbt` (it's soo fast!)

**Comment by olafurpg** [comment]
Thanks for reporting! If I understand correctly, you want to avoid clicking on the "import changes" button when changing the build in this dialogue here?

![screenshot 2019-01-11 at 16 57 17](https://user-images.githubusercontent.com/1408093/51044685-f8e2df00-15c1-11e9-804e-f479ee27dbb8.png)

My personal view is that this feature is fairly high-risk and low-value. I'm happy to be convinced otherwise. Some thoughts:

- for the subset of users who enable it, it saves you a few seconds from clicking on "import changes" every time you change the build.
- when the option is enabled, you may occasionally have to spend a few seconds to cancel the automatic import because you need make more changes.
- the setting needs to be implemented/maintained in the codebase and documented on the website so interested users can discover it.

The sbt import step takes at least 10 seconds in my experience, even for the simplest builds with minimal plugins. I would be more willing to consider this feature if the step took only ~1-2 seconds.

**Comment by kubukoz** [comment]
Yes, that's what I had in mind.

These are fair reasons not to add this, I agree. How about a keyboard shortcut for importing that also kills this window in case it appears?

**Comment by gabro** [comment]
@kubukoz I think you can add a shortcut in VSCode yourself, since import build is a command

**Comment by kubukoz** [comment]
Ah, wonderful. I'll try that.

**Comment by gabro** [comment]
I'll close this. If someone has any suggestion on how to make this feature less high-risk or more low-value (borrowing from @olafurpg definition above), feel free to comment and we can consider re-opening.

**Comment by nsadeh** [comment]
Hi @gabro , just seeing this looking for auto-import for Metals. This feature is available in IntelliJ.

Is there a way to make the import terminate if another import is queued? I believe that's how it's done in IntelliJ. For example

1. User makes a change => import A starts
2. Before the import complete, User makes another change => import B starts, import A canceled

I believe that gets rid of the risk you mention.

**Comment by gabro** [comment]
@nsadeh thanks for the suggestion. I think it partially addresses the issue, yes, although I'm still not convinced the effort in nailing the UX is worth the gain.

@olafurpg any additional thought on this?

**Comment by olafurpg** [comment]
I’m still not convinced that auto import is worth implementing. One feature which could help is to make it easier to bind a keyboard shortcut for the “import build” command. The shortcut could be displayed in the message request to improve discovery (this is how it works in IntelliJ). This would require an editor specific solution however.

**Comment by nsadeh** [comment]
Besides the keybinding, IntelliJ also has an option “enable auto import” every time it prompts for import. It you click it, every change will get auto imported. I find that helpful. 

It’s ok if you don’t - It’s your product. I just commented to make sure you know that along with the original filer of this issue some people find this feature helpful at reducing friction. I have made a key binding in VSCode for this and it reduces friction somewhat tho not to the extent of auto-import.
